tofu/geom/_core.py

```diff
@@ -5760,6 +5760,86 @@ def calc_min_rho_from_Plasma2D(self, plasma, t=None, log='min',
                                     for vv in np.split(val, lind, axis=-1)])
         return vals, pts, t
 
+    def get_inspector(self, ff):
+        out = inspect.signature(ff)
+        pars = out.parameters.values()
+        na = np.sum([(pp.kind == pp.POSITIONAL_OR_KEYWORD
+                      and pp.default is pp.empty) for pp in pars])
+        kw = [pp.name for pp in pars if (pp.kind == pp.POSITIONAL_OR_KEYWORD
+                                         and pp.default is not pp.empty)]
+        return na, kw
+
+    def check_ff(self, ff, t=None, ani=None):
```

Comment:
All good to me, the more detailed the error messages, the better it is for everyone indeed  :-)